### PR TITLE
Fix inconsistent partial model naming in Modelica.Magnetic

### LINKED ISSUE #496: Inconsistent partial model naming in Modelica.Magnetic
**Reported by christiankral on 4 Feb 2011 14:18 UTC**
The partial models 
 * `Interfaces.PartialTwoPortsElementary` and 
 * `Interfaces.PartialTwoPorts` 
are implemented differently in the packages 
 * `Modelica.Magnetic.FluxTubes` and 
 * `Modelica.Magnetic.FundamentalWave`.

# Current Implementation
## `FluxTubes.Interfaces.PartialTwoPortsElementary`

```
partial model PartialTwoPortsElementary 
  PositiveMagneticPort port_p;
  NegativeMagneticPort port_n;
end PartialTwoPortsElementary;

```

## `FluxTubes.Interfaces.PartialTwoPorts`

```
partial model PartialTwoPorts 
  extends FluxTubes.Interfaces.PartialTwoPortsElementary;
  SI.MagneticPotentialDifference V_m;
  SI.MagneticFlux Phi;

equation 
  V_m = port_p.V_m - port_n.V_m;
  Phi = port_p.Phi;
  0 = port_p.Phi + port_n.Phi;
end PartialTwoPorts;
```

## `FundamentalWave.Interfaces.PartialTwoPort`
```
partial model PartialTwoPort 
  PositiveMagneticPort port_p;
  NegativeMagneticPort port_n;
end PartialTwoPort;
```

## `FundamentalWave.Interfaces.PartialTwoPortElementary`
```
partial model PartialTwoPortElementary
  extends FundamentalWave.Interfaces.PartialTwoPort;
  Modelica.SIunits.ComplexMagneticPotentialDifference V_m;
  Modelica.SIunits.ComplexMagneticFlux Phi;

equation 
  V_m = port_p.V_m - port_n.V_m;
  port_p.Phi = Phi;
  port_p.Phi + port_n.Phi = Complex(0,0);
end PartialTwoPortElementary;
```

The naming of the partial models in the fundamental wave library is simply a bug. 

# Additional Issues

The term `port` used in the magnetic and thermal domain refers to a connector and it represents two pins or plugs in the electrical domains. We should come up with some clarification here. We should also unify whether or not we should used the term `Partial` in the class name. We have various different implementations in the MSL. 

The implementation of partial models is even entirely different throughout the package `Modelica.Electrical`: 

 * `Analog.Interfaces.TwoPin` is a partial model containing 
   * definition of voltage drop `v` across pins  
 * `MultiPhase.Interfaces.TwoPlug` is a partial model containing
   * definition of voltage drop `v` across plugs
   * definition of current `i` into positive plug
 * `Analog.Interfaces.OnePort` and `MultiPhase.Interfaces.OnePort` are partial models with two pins or plugs, respectively [the problem here is that we are using the term `port` as an abstraction for two pins and plugs, respectively, whereas this term is used differently, e.g, in the magnetic domain]; these partial models contain 
   * definition of voltage drop `v` across pins and plugs, respectively 
   * definition of current `i` into positive pin and plug, respectively
   * current balance of positive and negative pin (or plug)
 * `Analog.Interfaces.TwoPort` and `MultiPhase.Interfaces.TwoPort` are partial models with four pins or plugs, respectively; these partial models contain 
   * definition of voltage drop `v` across pins and plugs, respectively 
   * definition of current `i` into positive pin (or plug)
   * current balance of positive and negative pin (or plug)
 * `MultiPhase.Interfaces.FourPlug` contains
   * definition of voltage drop `v` across paris of plugs, respectively 
   * definition of current `i` into positive plugs

However, a unification of namings and implementations is highly desirable since all the differences are confusing to experienced and news users. I understand that we do need a conversion script for renaming partial models; so we do have wait for Modelica 4, I suppose.

# Proposal

So here we have lot very different namings with different implementations. On my opinion we need three different partial model types:
 * `TwoPinGeneric`, `TwoPlugGeneric`, etc., which contains only the connectors; these are the most generic components; they are also used for sources and sensors since it may be undesired in these cases to have a pre defined variable for either the potential difference or the flow variable
 * `TwoPinElementary`, `TwoPlugElementary`, etc., which contains the connectors and the definitions of the potential difference, e.g., `v`, and the flow variable into the positive connector, e.g., `i`; these models are mostly used for graphical programming  
 * `TwoPin`, `TwoPlug`, or `TwoPinBalanced`, `TwoPlugBalanced`, etc., containing the flow balances in addition to the `Elemenary` named partial models; these models are used for textual programming

Naming may be discussed.

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/496

### PR DESCRIPTION
**Pull request by christiankral**
close #496

### CODE REVIEW

File: ModelicaTestConversion4.mo [hunk 1]
```diff
@@ -675,5 +831,5 @@ Conversion test for <a href=\"https://github.com/modelica/ModelicaStandardLibrar
 </html>"));
     end Issue2944;
   end SIunits;
-  annotation(uses(Modelica(version="3.2.3")));
+  annotation(uses(Modelica(version="3.2.3"), Complex(version="3.2.2")));
```
Comment (dietmarw) [inline ModelicaTestConversion4.mo]: You might want to remove this accidental change.

**Comment by christiankral** [comment]
Yes, I will include it in this PR in the next couple days: 

* I will remove the "Partial" from the names 
* I will leave the remaining names as they are

The partial heat port actually will be covered in #359

**Comment by christiankral** [comment]
Grrrmpf! Where do these conflicts come from? I do now how to deal with that. @beutlich Please help again -- this somehow happened after I force-pushed a wrong commit message through GitKraken: what did I wrong?

**Comment by beutlich** [comment]
Will do. Hang on.

**Comment by beutlich** [comment]
>  @beutlich Please help again

Done.

**Comment by christiankral** [comment]
Thanks a lot!

**Comment by AHaumer** [comment]
Merging that PR, wouldn't we spill the big amount of work done for renaming MulitPhase to Polyphase, AsynchronousInductionMachine to InductionMachine and so on?

**Comment by beutlich** [comment]
> 
> 
> Merging that PR, wouldn't we spill the big amount of work done for renaming MulitPhase to Polyphase, AsynchronousInductionMachine to InductionMachine and so on?

No, all save to merge, since I rebased and resolved the conflicts (several times).

**Pull request merged**